diff --git a/sunrpc/netname.c b/sunrpc/netname.c
index bf7f0b81c4..2ee06f2b8d 100644
--- a/sunrpc/netname.c
+++ b/sunrpc/netname.c
@@ -20,11 +20,11 @@
 #include <string.h>
 #include <rpc/rpc.h>
 #include <shlib-compat.h>
+#include <libc-diag.h>
 
 #include "nsswitch.h"
 
 #define	OPSYS_LEN 4
-#define	MAXIPRINT (11)		/* max length of printed integer */
 static const char OPSYS[] = "unix";
 
 int
@@ -32,7 +32,7 @@ user2netname (char netname[MAXNETNAMELEN + 1], const uid_t uid,
 	      const char *domain)
 {
   char dfltdom[MAXNETNAMELEN + 1];
-  size_t i;
+  ssize_t i;
 
   if (domain == NULL)
     {
@@ -45,11 +45,15 @@ user2netname (char netname[MAXNETNAMELEN + 1], const uid_t uid,
       dfltdom[MAXNETNAMELEN] = '\0';
     }
 
-  if ((strlen (dfltdom) + OPSYS_LEN + 3 + MAXIPRINT) > (size_t) MAXNETNAMELEN)
+  i = snprintf (netname, MAXNETNAMELEN + 1, "%s.%d@%s", OPSYS, uid, dfltdom);
+  if (i <= 0 || i > (ssize_t) MAXNETNAMELEN)
     return 0;
 
-  sprintf (netname, "%s.%d@%s", OPSYS, uid, dfltdom);
-  i = strlen (netname);
+  /* GCC with -Os warns that sprint might overflow while handling dfltdom,
+     however the above test does check if an overflow would happen.  */
+  DIAG_PUSH_NEEDS_COMMENT;
+  DIAG_IGNORE_Os_NEEDS_COMMENT (8, "-Wformat-overflow");
+  DIAG_POP_NEEDS_COMMENT;
   if (netname[i - 1] == '.')
     netname[i - 1] = '\0';
   return 1;
